src/core/common/message.cpp

```diff
@@ -148,6 +147,10 @@ Buffer *MessagePool::NewBuffer(uint8_t aPriority)
     {
         otLogInfoMem("No available message buffer");
     }
+    else
+    {
+        buffer->SetNextBuffer(NULL);
```

Comment:
why moving it here?

Reply:
Always clean next even if the buffer is allocated by platform.